spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClient.java

```diff
@@ -67,13 +68,10 @@ public String description() {
 
 	private void addInstancesToList(List<ServiceInstance> instances, String serviceId, QueryParams queryParams) {
 		HealthServicesRequest.Builder requestBuilder = HealthServicesRequest.newBuilder()
-				.setPassing(this.properties.isQueryPassing()).setQueryParams(queryParams)
-				.setToken(this.properties.getAclToken());
-		String queryTag = this.properties.getQueryTagForService(serviceId);
-		if (queryTag != null) {
-			requestBuilder.setTag(queryTag);
-		}
+				.setPassing(properties.isQueryPassing()).setQueryParams(queryParams).setToken(properties.getAclToken());
+		Optional.ofNullable(properties.getQueryTagsForService(serviceId)).ifPresent(requestBuilder::setTags);
```

Comment:
We don't use `Optional` in place of if statements, please update.

Reply:
K. Will do. I can see in this simple case that this could be an "abuse" of an Optional (aka replacing a simple if w/ it). 

What is the stance when accessing an N level, nullable property path such as `user.contactInfo.address.street`:

**If statement**
```java
User.contactInfo.address.street
if (user != null && user.getContactInfo() != null && user.getContactInfo().getAddress() != null) {
  return user.getContactInfo().getAddress().getStreet();
}
```
vs. using `Optional + map` such as:
```java
return Optional.ofNullable(user)
   .map(User::contactInfo)
   .map(ContactInfo::getAddress)
   .map(Address::street)
    .orElse(null);
```
In groovy and other langs this would be as simple as `user?.contactInfo?.address?.street` of course. 

Not asking to spark a debate or push any preference - just want to know if this is a hard rule or not.